### PR TITLE
Implement conversion from ASE to ``System``

### PR DESCRIPTION
**Pull request by frostedoyster**
# Contributor (creator of pull-request) checklist

 - [x] Tests updated (for new features and bugfixes)?
 - [x] Documentation updated (for new features)?
 - [?] Issue referenced (for PRs that solve an issue)?

# Reviewer checklist

 - [ ] CHANGELOG updated with public API or any other important changes?

----
📚 Documentation preview 📚: https://metatensor--534.org.readthedocs.build/en/534/

### CODE REVIEW

File: python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+try:
+    import ase
+
+    HAS_ASE = True
+except ImportError:
+    HAS_ASE = False
+
+
+class IntoSystem:
```
Comment (Luthaf) [inline python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py:16]: This should have a docstring and be added to the documentation

File: python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                dtype=dtype,
+                device=device,
+            ),
+            cell=torch.tensor(
+                system.cell[:],
+                requires_grad=cell_requires_grad,
+                dtype=dtype,
+                device=device,
+            ),
+            types=torch.tensor(system.numbers, device=device),
```
Comment (Luthaf) [inline python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py:95]:
```suggestion
            types=torch.tensor(system.numbers, device=device, dtype=torch.int32),
```

File: python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    if isinstance(system, ase.Atoms):
+        return System(
+            positions=torch.tensor(
+                system.positions,
+                requires_grad=positions_requires_grad,
+                dtype=dtype,
+                device=device,
+            ),
+            cell=torch.tensor(
+                system.cell[:],
```
Comment (Luthaf) [inline python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py:90]: This should check if pbc are enabled, and use a zero cell matrix if they are not. Also, this should warn/error if the pbc are different for different axes.

File: docs/src/atomistic/reference/systems.rst [hunk 4]
```diff
@@ -7,3 +7,5 @@ Systems
 .. autoclass:: metatensor.torch.atomistic.NeighborsListOptions
     :members:
     :special-members: __eq__, __ne__
+
+.. autofunction:: metatensor.torch.atomistic.systems_to_torch
```
Comment (Luthaf) [inline docs/src/atomistic/reference/systems.rst:11]: we should edit the docstring of `System` that is currently pointing to rascaline to point to this one instead

**Review by Luthaf — CHANGES_REQUESTED** [review]

File: python/metatensor-torch/metatensor/torch/atomistic/documentation.py [hunk 5]
```diff
@@ -23,8 +23,12 @@ def __init__(
 
         .. admonition:: Converting data to metatensor `System`
 
-            Some external packages provides ways to create :py:class:`System` using data
-            from other libraries:
+            ``metatensor.torch.atomistic`` provides a way to convert ``ase.Atoms``
+            instances to :py:class:`System` using the :py:func:`systems_to_torch()`
+            function.
```
Comment (Luthaf) [inline python/metatensor-torch/metatensor/torch/atomistic/documentation.py:28]:
```suggestion
            We provides a way to convert :py:class:`ase.Atoms` to :py:class:`System` 
            using :py:func:`metatensor.torch.atomistic.systems_to_torch`.
```

File: python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+except ImportError:
+    HAS_ASE = False
+
+
+class IntoSystem:
+    """A type that can be converted into a ``metatensor.torch.atomistic.System``.
+
+    This is an abstract class that is used to indicate a class whose objects
+    can be converted into a ``metatensor.torch.atomistic.System``. For the moment,
+    the only supported type is ``ase.Atoms``."""
```
Comment (Luthaf) [inline python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py:22]:
```suggestion
    the only supported type is :py:class:`ase.Atoms`."""
```

File: python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+try:
+    import ase
+
+    HAS_ASE = True
+except ImportError:
+    HAS_ASE = False
+
+
+class IntoSystem:
+    """A type that can be converted into a ``metatensor.torch.atomistic.System``.
```
Comment (Luthaf) [inline python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py:18]:
```suggestion
    """A type that can be converted into a :py:class:`metatensor.torch.atomistic.System`.
```

File: python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+    This is an abstract class that is used to indicate a class whose objects
+    can be converted into a ``metatensor.torch.atomistic.System``. For the moment,
```
Comment (Luthaf) [inline python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py:21]:
```suggestion
    can be converted into a :py:class:`metatensor.torch.atomistic.System`. For the moment,
```

Might also work like this

```suggestion
    can be converted into a :py:class:`System`. For the moment,
```

File: python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+    if np.all(system.pbc):
+        cell = torch.tensor(
+            system.cell[:],
+            requires_grad=cell_requires_grad,
+            dtype=dtype,
+            device=device,
+        )
+    elif not np.any(system.pbc):
+        cell = np.zeros((3, 3))
```
Comment (Luthaf) [inline python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py:108]: if the cell is not already zero, we send a warning in rascaline (to not modify the user data without their knowledge). Should we do the same here?

Comment (frostedoyster) [reply python/metatensor-torch/metatensor/torch/atomistic/systems_to_torch.py:108]: Sure

**Review by Luthaf — APPROVED** [review]

**Pull request merged**